exonum/src/crypto/mod.rs

```diff
@@ -76,7 +41,24 @@ use helpers::Round;
 #[macro_use]
 mod macros;
 
-pub mod x25519;
+pub(crate) mod crypto_lib;
+// A way to set an active cryptographic backend is to export it as `crypto_impl`.
+#[cfg(feature = "sodiumoxide-crypto")]
+use self::crypto_lib::sodiumoxide as crypto_impl;
+
+#[cfg(feature = "sodiumoxide-crypto")]
+pub use self::crypto_lib::sodiumoxide::x25519;
+
+#[doc(inline)]
+pub use self::crypto_impl::HASH_SIZE;
```

Comment:
I prefer to group usings the following way:
```rust
pub use self::crypto_impl::{HASH_SIZE, PUBLIC_KEY_LENGTH, ...};
```
Not sure if it will work with the `#[doc(inline)]` attribute.